Plots/XY/NCL_xy_18.py

```diff
@@ -0,0 +1,124 @@
+
+nfiles = ["../../data/netcdf_files/TREFHT.B06.66.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.67.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.68.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.69.atm.1890-1999ANN.nc"]
+nds = xr.open_mfdataset(nfiles, concat_dim='case', combine='nested',
+                        preprocess=assume_noleap_calendar, decode_times=False)
+
+vfiles = ["../../data/netcdf_files/TREFHT.B06.61.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.59.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.60.atm.1890-1999ANN.nc",
+          "../../data/netcdf_files/TREFHT.B06.57.atm.1890-1999ANN.nc"]
+vds = xr.open_mfdataset(vfiles, concat_dim='case', combine='nested',
+                        preprocess=assume_noleap_calendar, decode_times=False)
+
+gds = xr.open_dataset("../../data/netcdf_files/gw.nc")
+gds = gds.expand_dims(dim={'lon': nds.lon})
+
+###############################################################################
+# Some parameters
+nyrs = len(nds["time"])
+nlon = len(nds["lon"])
+nlat = len(nds["lat"])
+time =
+
+###############################################################################
+# OBS
+
+obs = np.loadtxt("../../data/ascii_files/jones_glob_ann_2002.asc", dtype=float)
+
+###############################################################################
+# NCL-based Weighted Mean Function
+
+def horizontal_weighted_mean(var, wgts):
+    return (var * wgts).sum(dim=['lat', 'lon']) / wgts.sum(dim=['lat', 'lon'])
+
+###############################################################################
+# NATURAL
+
+gavn = horizontal_weighted_mean(nds["TREFHT"], gds["gw"])
+gavan = gavn - gavn.isel(time=slice(0,30)).mean(dim='time')
+
+###############################################################################
+# ALL
+
+gavv = horizontal_weighted_mean(vds["TREFHT"], gds["gw"])
+gavav = gavv - gavv.isel(time=slice(0,30)).mean(dim='time')
+
+###############################################################################
+# CALCULATE ENSEMBLE MIN & MAX
+
+gavan_min = gavan.min(dim='case')
+gavan_max = gavan.max(dim='case')
+gavan_avg = gavan.mean(dim='case')
+
+gavav_min = gavav.min(dim='case')
+gavav_max = gavav.max(dim='case')
+gavav_avg = gavav.mean(dim='case')
+
+obs_avg = obs[34:144] - np.mean(obs[34:64])
+
+###############################################################################
+# Create plot
+
+fig, ax = plt.subplots(figsize=(10.5, 6))
+
+ax.tick_params(labelsize="small")
+ax.minorticks_on()
+ax.xaxis.set_minor_locator(tic.AutoMinorLocator(n=4))
+ax.yaxis.set_minor_locator(tic.AutoMinorLocator(n=3))
+ax.tick_params(axis="both", labelsize=20)
+ax.tick_params("both", length=8, width=1.50, which="major", bottom=True, top=True, left=True, right=True)
+ax.tick_params("both", length=5, width=0.75, which="minor", bottom=True, top=True, left=True, right=True)
+
+time = [t.year for t in gavan.time.values]
```

Comment:
`time = gavan.time.dt.year  # Use xarray's DatetimeAccessor for convenience`

Reply:
Oh, nice!  I'm not familiar with that, but that's awesome.

Reply:
Done.